pkg/controller/finalizer.go

```diff
@@ -91,12 +92,12 @@ func (c *Controller) hasFinalizer(ctx context.Context, obj interface{}) (bool, e
 		return false, microerror.Mask(err)
 	}
 	finalizerName := getFinalizerName(c.name)
-	selfLink := accessor.GetSelfLink()
+	uid := string(accessor.GetUID())
```

Comment:
We now use UID as the cache key. It doesn't have a string function so just casting it.

https://github.com/kubernetes/apimachinery/blob/v0.20.5/pkg/types/uid.go#L19-L22